packages/shared/src/components/drawers/NavDrawerItem.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { ReactElement } from 'react';
+import { Button, ButtonProps, ButtonVariant } from '../buttons/Button';
+
+export interface NavItemProps
+  extends Pick<ButtonProps<'a'>, 'href' | 'icon' | 'onClick'> {
+  label: string;
+  isHeader?: boolean;
+}
+
+const NavDrawerHeaderItem = ({ label }: { label: string }): ReactElement => (
+  <h3 className="my-2 flex h-12 items-center font-bold typo-callout first:mt-0">
+    {label}
+  </h3>
+);
+
+export function NavDrawerItem({
+  icon,
+  isHeader = false,
+  href,
+  label,
+  onClick,
+}: NavItemProps): ReactElement {
+  if (isHeader) {
+    return <NavDrawerHeaderItem label={label} />;
+  }
+
+  return (
+    <Button
+      role="menuitem"
+      type="button"
+      icon={icon}
+      className="!justify-start !font-normal"
+      onClick={onClick}
+      href={href}
+      tag={href ? 'a' : 'button'}
+      variant={ButtonVariant.Tertiary}
```

Comment:
Should these render as button variants?
Seems like it's not actually a button, more a styled element? (doesn't match our button guidelines basically)

Reply:
Maybe double check what we do in sidebar currently, but think it should not abuse the buttons basically looking at figma.

Reply:
I did a preview of the component in Figma, and it looks exactly like a button to me.

<img width="358" alt="Screenshot 2024-03-13 at 08 45 41" src="https://github.com/dailydotdev/apps/assets/9974711/500c513a-50f9-4bd6-835d-72a06a410d6a">

Reply:
In sidebar we use `<li` tags with an `<a href` inside, but in this case, the buttons already have most of the styling we need. Out button component also supports rendering an `<a href`, so not sure why you think this would be abuse?

Reply:
Created a new button variant `Option` for now. Also created a followup task to clean this up: https://dailydotdev.atlassian.net/browse/MI-231